libcodechecker/analyze/log_parser.py

```diff
@@ -93,14 +93,28 @@ def parse_compile_commands_json(logfile, add_compiler_defaults=False):
     counter = 0
     for entry in data:
         sourcefile = entry['file']
+
+        if not os.path.isabs(sourcefile):
+            # Newest versions of intercept-build can create the 'file' in the
+            # JSON Compilation Database as a relative path.
+            sourcefile = os.path.join(os.path.abspath(entry['directory']),
+                                      sourcefile)
+
         lang = option_parser.get_language(sourcefile[sourcefile.rfind('.'):])
 
         if not lang:
             continue
 
         action = build_action.BuildAction(counter)
 
-        command = entry['command']
+        if 'command' in entry:
+            command = entry['command']
+        elif 'arguments' in entry:
+            # Newest versions of intercept-build create an argument vector
+            # instead of a command string.
+            command = ' '.join(entry['arguments'])
```

Comment:
For the record --- I've already removed the question because I found the solution meanwhile, the question was:

> How would you pass an argument list to clang, as right now the `build.json` never directly reaches the analyzer.

The build database is parsed and filtered, and from that do we create the analyzer invocation.

I've investigated the issue and came up with the solution that works both for _intercept-build_ and _ld-logger_, while also esentially fixing the aforementioned issue #505. Please see #631.
#631